pkg/workceptor/workceptor_test.go

```diff
@@ -108,3 +131,337 @@ func TestAllocateUnit(t *testing.T) {
+		expectedCalls func()
+	}{
+		{
+			name:      "get client tls config error",
+			tlsClient: "something",
+			errorMsg:  "terminated",
+			expectedCalls: func() {
+				mockNetceptor.EXPECT().GetClientTLSConfig(gomock.Any(), gomock.Any(), gomock.Any()).Return(&tls.Config{}, errors.New("terminated"))
+			},
+		},
+		{
+			name:          "sending secrets over non tls connection error",
+			tlsClient:     "",
+			params:        map[string]string{"secret_": "secret"},
+			errorMsg:      "cannot send secrets over a non-TLS connection",
+			expectedCalls: func() {},
+		},
+		{
+			name:          "invalid duration error",
+			tlsClient:     "",
+			ttl:           "ttl",
+			errorMsg:      "time: invalid duration \"ttl\"",
+			expectedCalls: func() {},
+		},
+		{
+			name:          "normal case",
+			tlsClient:     "",
+			ttl:           "1.5h",
+			errorMsg:      "",
+			signWork:      true,
+			expectedCalls: func() {},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.expectedCalls()
+			_, err := w.AllocateRemoteUnit("", "", tc.tlsClient, tc.ttl, tc.signWork, tc.params)
+
+			if tc.errorMsg != "" && tc.errorMsg != err.Error() && err != nil {
+				t.Errorf("expected: %s, received: %s", tc.errorMsg, err)
+			}
+
+			if tc.errorMsg == "" && err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestUnitStatus(t *testing.T) {
+	_, _, w := testSetup(t)
+	activeUnitsIDs := w.ListKnownUnitIDs()
+
+	_, err := w.UnitStatus(activeUnitsIDs[0])
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestCancelUnit(t *testing.T) {
+	_, _, w := testSetup(t)
+	activeUnitsIDs := w.ListKnownUnitIDs()
+
+	err := w.CancelUnit(activeUnitsIDs[0])
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestReleaseUnit(t *testing.T) {
+	_, _, w := testSetup(t)
+	activeUnitsIDs := w.ListKnownUnitIDs()
+
+	err := w.ReleaseUnit(activeUnitsIDs[0], true)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestListKnownUnitIDs(t *testing.T) {
+	t.Parallel()
+	_, _, w := testSetup(t)
+	testCases := []struct {
+		name     string
+		workType string
+	}{
+		{
+			name: "parallel test 1",
+		},
+		{
+			name: "parallel test 2",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			w.ListKnownUnitIDs()
+		})
+	}
+}
+
+func TestGetResults(t *testing.T) {
+	_, _, w := testSetup(t)
+	activeUnitsIDs := w.ListKnownUnitIDs()
+
+	testCases := []struct {
+		name          string
+		unitID        string
+		expectedCalls func()
+		hasError      bool
+	}{
+		{
+			name:          "could not file unit error",
+			unitID:        "",
+			hasError:      true,
+			expectedCalls: func() {},
+		},
+		{
+			name:          "unit stdout file does not exist",
+			unitID:        activeUnitsIDs[0],
+			expectedCalls: func() {},
+		},
+		{
+			name:   "unit stdout file exists",
+			unitID: activeUnitsIDs[0],
+			expectedCalls: func() {
+				file := fmt.Sprintf("/tmp/test/%s/stdout", activeUnitsIDs[0])
+				f, err := os.Create(file)
```

Comment:
The os calls need to be completely removed from the tests, I have some examples on how I did that that can be reused.

Reply:
https://github.com/ansible/receptor/blob/devel/pkg/workceptor/stdio_utils.go#L15